src/ol/format/WKB.js

```diff
@@ -0,0 +1,866 @@
+      const writerLUT = {
+        [GeometryType.POINT]: this.writePoint,
+        [GeometryType.LINE_STRING]: this.writeLineString,
+        [GeometryType.POLYGON]: this.writePolygon,
+        [GeometryType.MULTI_POINT]: this.writeMultiPoint,
+        [GeometryType.MULTI_LINE_STRING]: this.writeMultiLineString,
+        [GeometryType.MULTI_POLYGON]: this.writeMultiPolygon,
+      };
+      writerLUT[geomType].call(this, geom.getCoordinates(), geom.getLayout());
+    } else if (geom instanceof GeometryCollection) {
+      this.writeGeometryCollection(geom.getGeometriesArray());
+    }
+  }
+
+  getBuffer() {
+    const byteLength = this.writeQueue.reduce((acc, item) => acc + item[0], 0);
+    const buffer = new ArrayBuffer(byteLength);
+    const view = new DataView(buffer);
+
+    let pos = 0;
+    this.writeQueue.forEach((item) => {
+      switch (item[0]) {
+        case 1:
+          view.setUint8(pos, item[1]);
+          break;
+        case 4:
+          view.setUint32(pos, item[1], this.isLittleEndian);
+          break;
+        case 8:
+          view.setFloat64(pos, item[1], this.isLittleEndian);
+          break;
+        default:
+          break;
+      }
+
+      pos += item[0];
+    });
+
+    return buffer;
+  }
+}
+
+/**
+ * @typedef {Object} Options
+ * @property {boolean} [splitCollection=false] Whether to split GeometryCollections into multiple features on reading.
+ * @property {boolean} [hex=true] Returns hex string instead of ArrayBuffer for output.
+ * @property {boolean} [littleEndian=true] Use littleEndian for output.
+ * @property {boolean} [ewkb=true] Use EWKB format for output.
+ * @property {import("../geom/GeometryLayout").default} [geometryLayout=null] Use specific coordinate layout for output features (null: auto detect)
+ * @property {number} [nodataZ=0] If the `geometryLayout` doesn't match with geometry to be output, this value is used to fill missing coordinate value of Z.
+ * @property {number} [nodataM=0] If the `geometryLayout` doesn't match with geometry to be output, this value is used to fill missing coordinate value of M.
+ * @property {number|boolean} [SRID=true] SRID for output. Specify integer value to enforce the value as a SRID. Specify `true` to extract from `dataProjection`. `false` to suppress the output. This option only takes effect when `ewkb` is `true`.
+ */
+
+/**
+ * @classdesc
+ * Geometry format for reading and writing data in the `Well-Known Binary` (WKB) format.
+ * Also supports `Extended Well-Known Binary` (EWKB) format, used in PostGIS for example.
+ *
+ * @api
+ */
+class WKB extends FeatureFormat {
+  /**
+   * @param {Options} [opt_options] Optional configuration object.
+   */
+  constructor(opt_options) {
+    super();
+
+    const options = opt_options ? opt_options : {};
+
+    this.splitCollection = Boolean(options.splitCollection);
+
+    this.viewCache_ = null;
+
+    this.hex = options.hex !== false;
+    this.littleEndian = options.littleEndian !== false;
```

Comment:
Minor, but if these names are suffixed with `_` (e.g. `littleEndian_`) then they will be available for minification.

Reply:
Thanks! I renamed all of them.